Userland/Libraries/LibWeb/Layout/InitialContainingBlock.h

```diff
@@ -8,6 +8,7 @@
 
 #include <LibWeb/DOM/Document.h>
 #include <LibWeb/Layout/BlockContainer.h>
+#include <LibWeb/Layout/LayoutPosition.h>
```

Comment:
Not sure why this include is needed in the .h? It's only used as a `const&` here, does it suffice to be in the .cpp instead?

Reply:
It includes Document.h which includes ParentNode.h which includes Node.h which used to include LayoutPosition.h. This used to be the only path through which `Layout::LayoutPosition` got declared. Perhaps a declaration is sufficient, but it definitely needs to be declared, at minimum.

It also needs a definition of `Layout::LayoutRange`, so that the method `LayoutRange const& InitialContainingBlock::selection() const` knows how to arrange the registers. `Layout::LayoutRange` is defined in `LayoutPosition.h`, so we need to include that header.

In theory you could split that header into two, but it is very short (i.e. cannot significantly shorten the compile time, it's barely even 60 lines!), and was included anyway (i.e. this PR does not increase the number of included headers). You are very welcome to shorten compile times! I'd like to do something different in this PR, namely remove cyclic includes.

Reply:
My dude, I asked just because it is not obvious why a header is needed when a forward declaration would surely suffice. It's a mistake that this type isn't in Forward.h already, but yes that can be another PR.

Reply:
:+1: Great!

I don't think that a forward declaration will suffice, due to `LayoutRange`, but I'm looking forward to your PR :)